### PR TITLE
Non linear softening

### PR DESCRIPTION
**Pull request by albert-de-montserrat**
Adds non linear softening from [Thibault et al 21](https://agupubs.onlinelibrary.wiley.com/doi/pdfdirect/10.1029/2021GC009675)

### CODE REVIEW

**Comment by boriskaus** [comment]
why is this called `NoLinearSoftening` instead of `NonlinearSoftening`? Typo?

File: src/Softening/Softening.jl [hunk 1]
```diff
@@ -29,3 +30,22 @@ LinearSoftening(min_max_values::NTuple{2, T}, lo_hi::NTuple{2, T}) where T = Lin
     
     return fma((one(T) - softening_var), softening.slope, softening.min_value)
 end
+
+## Non linear softening 
+# (Thibault et al 2021; https://agupubs.onlinelibrary.wiley.com/doi/pdfdirect/10.1029/2021GC009675)
```
Comment (boriskaus) [inline src/Softening/Softening.jl:35]: Duretz et al.

File: src/ConstitutiveRelationships.jl [hunk 2]
```diff
@@ -62,7 +62,8 @@ export param_info,
     iselastic,
     AbstractSoftening,
     NoSoftening,
-    LinearSoftening
+    LinearSoftening,
+    NoLinearSoftening
```
Comment (boriskaus) [inline src/ConstitutiveRelationships.jl:66]: would be better to call it `NonlinearSoftening`

**Comment by albert-de-montserrat** [comment]
> why is this called `NoLinearSoftening` instead of `NonlinearSoftening`? Typo?

it is indeed a typo... i'm patching it

**Pull request merged**